lib/tasks/sitemap.rake

```diff
@@ -8,8 +10,35 @@ namespace :sitemap do
 
     output_directory = File.join(PROJECT_ROOT, "public")
     sitemap = Sitemap.new(output_directory)
-    sitemap.generate(SearchConfig.default_instance)
+    sitemap.generate_and_replace(SearchConfig.default_instance)
 
     sitemap.cleanup
   end
+
+  desc "Generate new sitemap files and upload to S3"
+  task :generate_and_upload do
+    @bucket_name = ENV['AWS_S3_BUCKET_NAME']
+    raise "Missing required AWS_S3_BUCKET_NAME" if @bucket_name.nil?
+
+    Dir.mktmpdir do |output_directory|
+      filenames = Sitemap.new(output_directory).generate(SearchConfig.default_instance)
+
+      filenames[:sitemaps].each do |filename, link_filename|
+        upload_to_s3(
+          "#{output_directory}/#{Sitemap::SUB_DIRECTORY}/#{filename}",
+          "#{Sitemap::SUB_DIRECTORY}/#{link_filename}"
+        )
+      end
+
+      upload_to_s3(
+        "#{output_directory}/#{Sitemap::SUB_DIRECTORY}/#{filenames[:index]}",
+        "sitemap.xml"
+      )
+    end
```

Comment:
Out of interest, why are there multiple sitemaps?

Reply:
Two reasons: firstly, there's a limit of how big a sitemap file can be (50,000 entries); and secondly, we've run out of memory previously trying to generate 50,000-entry sitemaps.

Reply:
Awesome, thanks for explaining 👍